packages/curve-ui-kit/src/shared/ui/ActionInfo.tsx

```diff
@@ -1,75 +1,159 @@
   large: 'bodyMRegular',
-}
+} as const satisfies Record<ComponentSize, TypographyVariantKey>
+
+const prevValueSize = {
+  small: 'bodySRegular',
+  medium: 'bodyMRegular',
+  large: 'bodyMRegular',
+} as const satisfies Record<ComponentSize, TypographyVariantKey>
 
-const addressSize: Record<ComponentSize, 'bodyXsBold' | 'highlightM' | 'headingSBold'> = {
+const valueSize = {
   small: 'bodyXsBold',
   medium: 'highlightM',
   large: 'headingSBold',
-}
+} as const satisfies Record<ComponentSize, TypographyVariantKey>
 
-const ActionInfo = ({ label, address, linkAddress, size = 'medium', copiedText }: ActionInfoProps) => {
+const ActionInfo = ({
+  label,
+  prevValue,
+  prevValueColor,
+  value,
+  valueColor,
+  valueLeft,
+  valueRight,
+  valueTooltip = '',
+  link,
+  size = 'medium',
+  copy = false,
+  copiedTitle,
+  loading = false,
+}: ActionInfoProps) => {
   const [isOpen, open, close] = useSwitch(false)
 
   const copyValue = () => {
-    copyToClipboard(address)
+    copyToClipboard(value)
     open()
   }
 
   return (
-    <Stack direction="row" alignItems="center" gap={Spacing.md} justifyContent="space-between">
-      <Typography variant={labelSize[size]} color="textSecondary">
+    <Stack direction="row" alignItems="center" gap={Spacing.sm}>
+      <Typography flexGrow={1} variant={labelSize[size]} color="textSecondary">
         {label}
       </Typography>
-      <Stack direction="row" alignItems="center">
-        <Typography variant={addressSize[size]} color="textPrimary" sx={{ marginRight: Spacing.sm }}>
-          {shortenAddress(address)}
-        </Typography>
-        <IconButton size="small" onClick={copyValue} color="primary">
-          <ContentCopy />
-        </IconButton>
-        <IconButton component={Link} href={linkAddress} target="_blank" rel="noopener" size="small" color="primary">
-          <CallMade />
-        </IconButton>
+
+      {prevValue && (
+        <Stack direction="row" alignItems="center">
+          <Typography variant={prevValueSize[size]} color={prevValueColor ?? 'textSecondary'}>
+            {prevValue}
+          </Typography>
+
+          <ArrowForwardIcon
+            sx={{
+              width: IconSize.sm,
+              height: IconSize.sm,
+              color: (t) => t.palette.text.primary,
+            }}
+          />
+        </Stack>
+      )}
+
+      <Stack direction="row" alignItems="center" gap={Spacing.xs}>
+        {valueLeft}
+
+        <WithSkeleton loading={!!loading}>
+          <Tooltip
+            title={valueTooltip}
+            placement="top"
+            slotProps={{
+              popper: {
+                sx: {
+                  userSelect: 'none',
+                  pointerEvents: 'none',
+                },
+              },
+            }}
+          >
+            <Typography variant={valueSize[size]} color={valueColor ?? 'textPrimary'}>
+              {!loading ? value : typeof loading === 'string' ? loading : MOCK_SKELETON}
```

Comment:
Double negatives are hard to read
```suggestion
              {typeof loading === 'string' ? loading : loading ? MOCK_SKELETON : value}
```